### ISSUE TITLE
LDAP Group Sync issue

### ISSUE THREAD

**Issue by spanchy** [RocketChat/Rocket.Chat#15673]
### Description:

When LDAP groups and "automatic removal of users from channels are synchronized" to "true", existing users in channels are deleted and added again + new employees. And so constantly, in a circle, every synchronization.

### Steps to reproduce:

1. go to settings LDAP > Sync and enable ldap group sync
2. Click set automatic removal of users from channels are synchronized to True
3. and see any channel for logs: User left channel and User joined to channel.

### Server Setup Information:

- Version of Rocket.Chat Server: 2.1.1
- Operating System: Debian Stretch 9.11
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: -
- NodeJS Version: 8.11.4
- MongoDB Version: 3.6.14

### Client Setup Information

- Desktop App or Browser Version: 2.15.5
- Operating System: Windows 10 x64

### Additional context
![duplicate_add_channel](https://user-images.githubusercontent.com/562733/67573332-bfd6be00-f740-11e9-93ad-9d061dc87018.JPG)
![scr_sec](https://user-images.githubusercontent.com/562733/67573333-c06f5480-f740-11e9-8e6f-cd8f1cd70e8c.JPG)
![scr_one](https://user-images.githubusercontent.com/562733/67573334-c06f5480-f740-11e9-85e7-636e690bca53.JPG)

**Comment by nklock** [comment]
I am also running into this issue. Every time the cron job kicks off to sync the user accounts, every room is flooded with dozens of users entering and leaving the channel. Can these notifications be disabled when the process is run in the background?

Or as a work around disable joining and leaving notifications altogether?

**Comment by Turbocube644** [comment]
Hi, I get the same Issue. Is there any new information?

**Comment by Turbocube644** [comment]
Hi,
the Issues still exists in 3.0.4. A fix would be much appreciated!

**Comment by spanchy** [comment]
> I am also running into this issue. Every time the cron job kicks off to sync the user accounts, every room is flooded with dozens of users entering and leaving the channel. Can these notifications be disabled when the process is run in the background?
> 
> Or as a work around disable joining and leaving notifications altogether?

Yes, you can disable this message in settings. Message -> Hide System message, and select what you want to hide.

**Comment by spanchy** [comment]
> Hi,
> I get the same Issue (Version 3.0.4).
> Is there any new information?

I wrote to support by email, and they replied that they would pass the information to the developers, but they haven 't fixed it yet.

**Comment by Examinato** [comment]
This issue still exists. It does persist only when a LDAP Group is synced to multiple channels. I have hidden the system messages, but the channel gets an unread count every sync because of these messages.

**Comment by Examinato** [comment]
This issue still exists. It does persist only when a LDAP Group is synced to multiple channels. I have hidden the system messages, but the channel gets an unread count every sync because of these messages.

**Comment by spanchy** [comment]
> This issue still exists. It does persist only when a LDAP Group is synced to multiple channels. I have hidden the system messages, but the channel gets an unread count every sync because of these messages.

yes, i have same problem.

**Comment by Examinato** [comment]
I also noticed that the notifications aren't working as it should with such a mixed channel:

On iOS mobile app
- when clicking notification, the right discussion isn't opened (this does work with normal groups/channels)
- the channel isn't on top, last activity is correct but it isn't sorted right

People were missing the messages in this mixed LDAP group channel, so I decided to only map one-on-one and create a manual channel with the people which I need to have together from multiple LDAP channels.

**Comment by EnCz** [comment]
We have the same problem, please take a look @MartinSchoeler

**Comment by liszca** [comment]
I can reproduce the message by hitting the Sync Button in the LDAP Configuration (GO: LDAP > Sync / Import >  "Execute Synchronization Now"):
![grafik](https://user-images.githubusercontent.com/32976532/106369571-e8d9fb80-6352-11eb-8e29-eac2db89c12e.png)

The list adds up every 10 Minutes as I do a Sync every 10 Minutes!

**Comment by neonardo1** [comment]
This is still an issue.

**Comment by liszca** [comment]
I guess it comes because two or more LDAP groups with useres are synced into a channel.

**Comment by MisterErwin** [comment]
Still an issue in 3.18.0-develop and 3.17.0
